public/app/features/scenes/components/layout/SceneGridLayout.tsx

```diff
@@ -0,0 +1,403 @@
+      };
+    });
+
+    return { lg, sm: lg.map((l) => ({ ...l, w: 24 })) };
+  }, [children]);
+
+  return (
+    <AutoSizer disableHeight>
+      {({ width }) => {
+        if (width === 0) {
+          return null;
+        }
+
+        // Dev only, to be removed
+        const background = generateGridBackground({
+          cellSize: { width: (width - 23 * GRID_CELL_VMARGIN) / 24, height: GRID_CELL_HEIGHT },
+          margin: [GRID_CELL_VMARGIN, GRID_CELL_VMARGIN],
+          cols: 24,
+          gridWidth: width,
+          theme,
+        });
+
+        return (
+          /**
+           * The children is using a width of 100% so we need to guarantee that it is wrapped
+           * in an element that has the calculated size given by the AutoSizer. The AutoSizer
+           * has a width of 0 and will let its content overflow its div.
+           */
+          <div style={{ width: `${width}px`, height: '100%', background }}>
+            <ReactGridLayout
+              width={width}
+              /*
+                  Disable draggable if mobile device, solving an issue with unintentionally
+                  moving panels. https://github.com/grafana/grafana/issues/18497
+                  theme.breakpoints.md = 769
+                */
+              isDraggable={width > 768}
+              isResizable={false}
+              containerPadding={[0, 0]}
+              useCSSTransforms={false}
+              margin={[GRID_CELL_VMARGIN, GRID_CELL_VMARGIN]}
+              cols={GRID_COLUMN_COUNT}
+              rowHeight={GRID_CELL_HEIGHT}
+              draggableHandle={`.grid-drag-handle-${model.state.key}`}
+              // @ts-ignore: ignoring for now until we make the size type numbers-only
+              layout={width > 768 ? layout.lg : layout.sm}
+              onDragStop={model.onDragStop}
+              onResizeStop={model.onResizeStop}
+              onLayoutChange={model.onLayoutChange}
+              isBounded={true}
+            >
+              {children.map((child) => {
+                return (
+                  // Grid items are flex, to be able to render flex layouts inside
+                  <div key={child.state.key} style={{ display: 'flex' }}>
+                    <child.Component model={child} key={child.state.key} />
+                  </div>
+                );
+              })}
+            </ReactGridLayout>
+          </div>
+        );
+      }}
+    </AutoSizer>
+  );
+}
+
+interface SceneGridRowState extends SceneLayoutChildState {
+  title: string;
+  isCollapsible?: boolean;
+  isCollapsed?: boolean;
+  children: Array<SceneObject<SceneLayoutChildState>>;
+}
+
+export class SceneGridRow extends SceneObjectBase<SceneGridRowState> {
+  static Component = SceneGridRowRenderer;
+  private _originalHeight = 0;
+
+  constructor(
+    state: Omit<SceneGridRowState, 'size'> & { size: Pick<GridCellLayout, 'x' | 'y' | 'height'> & { width?: number } }
+  ) {
+    super({
+      isResizable: true,
```

Comment:
@dprokop and I have been talking today about it. And it is not trivial when a row (containing a grid) is placed within a grid. A grid item, by definition, has a fixed amount of rows and cols to occupy. If you want to have a "row block" growing X number of grid rows, depending on the content, you need to modify the container size within the grid, changing the h property.

Looking at this complexity. We've re-thought if the row should display 100% of the context when the content height increment. We compare it to a text box within a canvas. If the text overflows, you go to the text box and resize it, but from the outer grid perspective, that element is meant to occupy X fixed amount of pixels. So, the same for the grid, if the user is resizing the content of the row, and the height grows, it needs to manually resize it. 

So we agreed that the rows should be resizable, the same as a sub-dashboard within a dashboard. And if the content grows, the container displays a scroll, and it is the user intentionally that goes and resizes it. Otherwise, if we want to have a growing element, this is why we have the flex layout.

WDYT @torkelo?

Reply:
Sounds good but it all depends on if we can solve nested grids and drags and drop between grids. If we can't we need to keep the current solution where the row is just a full width panel and the rows children are just panels in the top grid.

Reply:
Yep, we must solve it, otherwise, we won't be able to move panels between flex and grid layouts, and the experience will be quite limited.

We've tried today using the event bus, and it was working well so far. The changes are not pushed because we're missing updating the two grid states, but we're able to communicate them. 

We took the principle, there is only 1 panel dragging at a time, so all the grids subscribe to an element dragging, and store it in the state. In case that instance is called with the `onDrop`, it updates the state. When the drag stops, all `dragging` can be cleared. What we like about this approach is the complexity of moving between grids is only kept in the GridLayoutScene, and it is quite simple to debug and understand. Let's see how it works when we finish the implementation :)